src/browser/api.ts

```diff
@@ -247,6 +247,7 @@ const webApi: IPCApi = {
     executeBash: (workspaceId, script, options) =>
       invokeIPC(IPC_CHANNELS.WORKSPACE_EXECUTE_BASH, workspaceId, script, options),
     openTerminal: (workspacePath) => invokeIPC(IPC_CHANNELS.WORKSPACE_OPEN_TERMINAL, workspacePath),
+    generateTitle: (workspaceId) => invokeIPC(IPC_CHANNELS.WORKSPACE_GENERATE_TITLE, workspaceId),
```

Comment:
I don't think more IPC is necessary here. I'd rather keep the IPC surface area very small.

An example of how this adds complexity is it's not clear what sendMessageOptions (model, thinking, etc.) are being used to generate the title.